pysc2/replay_parsers/action_parser.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Action statistics parser for replays."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import six
+
+from pysc2.replay_parsers import base_parser
+
+class ActionParser(base_parser.BaseParser):
+  """Action statistics parser for replays."""
+
+  def __init__(self):
+    super(ActionParser, self).__init__()
+    self.camera_move = 0
+    self.select_pt = 0
+    self.select_rect = 0
+    self.control_group = 0
+    self.unit_ids = collections.defaultdict(int)
+    self.valid_abilities = collections.defaultdict(int)
+    self.made_abilities = collections.defaultdict(int)
+    self.valid_actions = collections.defaultdict(int)
+    self.made_actions = collections.defaultdict(int)
+
+  def merge(self, other):
+    """Merge another ReplayStats into this one."""
+    def merge_dict(a, b):
+      for k, v in six.iteritems(b):
+        a[k] += v
+    super(ActionParser,self).merge(other)      
+    self.camera_move += other.camera_move
+    self.select_pt += other.select_pt
+    self.select_rect += other.select_rect
+    self.control_group += other.control_group
+    merge_dict(self.unit_ids, other.unit_ids)
+    merge_dict(self.valid_abilities, other.valid_abilities)
+    merge_dict(self.made_abilities, other.made_abilities)
+    merge_dict(self.valid_actions, other.valid_actions)
+    merge_dict(self.made_actions, other.made_actions)
+
+  def valid_replay(self, info, ping):
+    """Make sure the replay isn't corrupt, and is worth looking at."""
```

Comment:
Changed valid replay conditions to be very low for test replay pass. The checks can stand as an example for users who wish to make valid_replay checks more aggressive.